src/scene/shader-lib/chunks/lit/frag/TBNderivative.js

```diff
@@ -20,6 +20,19 @@ void getTBN(vec3 tangent, vec3 binormal, vec3 normal) {
     // construct a scale-invariant frame
     float denom = max( dot(T,T), dot(B,B) );
     float invmax = (denom == 0.0) ? 0.0 : tbnBasis / sqrt( denom );
-    dTBN = mat3(T * invmax, -B * invmax, normal );
+
+    #ifdef TWO_SIDED_LIGHTING
+        #ifdef WEBGPU
+            dTBN = mat3(T * invmax, -B * invmax, gl_FrontFacing ? -normal : normal);
+        #else
+            dTBN = mat3(T * invmax, -B * invmax, gl_FrontFacing ? normal : -normal);
+        #endif
+    #else
+        #ifdef WEBGPU
+            dTBN = mat3(T * invmax, -B * invmax, -normal);
+        #else
+            dTBN = mat3(T * invmax, -B * invmax, normal);
+        #endif
+    #endif
```

Comment:
Webgpu lighting space should not be different to webgl, so this normal negation is terribly confusing.

Are textures inverted (upside down) on webgpu vs webgl @mvaligursky? (If so then tangents and binormals as calculated would be inverted).

I think we need to get to the bottom of why normals needs flipping here and fix it at the source. My suspicion is we have an inverted transform somewhere else in the graphics pipeline under webgpu.

Reply:
Actually I just noticed, looking in lit-shader.js, it seems we are already attempting to handle the normal for double-sided lighting case at https://github.com/playcanvas/engine/blob/main/src/scene/shader-lib/programs/lit-shader.js#L989. This calc must have a 🐛...

Reply:
You are right, it seems the issue is that it was handled in the wrong place. I have added a negatively scaled mesh to the example and I have gone though all variations of the shader to maker sure they all work correctly now.